kube-core/src/conversion/types.rs

```diff
@@ -0,0 +1,178 @@
+                req.types = Some(review.types);
+                Ok(req)
+            }
+            None => Err(ConvertConversionReviewError),
+        }
+    }
+}
+
+impl TryFrom<ConversionReview> for ConversionRequest {
+    type Error = ConvertConversionReviewError;
+
+    fn try_from(review: ConversionReview) -> Result<Self, Self::Error> {
+        ConversionRequest::from_review(review)
+    }
+}
+
+/// Part of ConversionReview which is set on output (i.e. generated by conversion webhook)
+#[derive(Serialize, Deserialize)]
+pub struct ConversionResponse {
+    /// Copied from the corresponding consructing [`ConversionRequest`].
+    /// This field is not part of the Kubernetes API, it's consumed only by `kube`.
+    #[serde(skip)]
+    pub types: Option<TypeMeta>,
+    /// Copy of .request.uid
+    pub uid: String,
+    /// Outcome of the conversion operation.
+    /// Success: all objects were successfully converted
+    /// Failure: at least one object could not be converted.
+    /// It is recommended that conversion fails as rare as possible.
+    pub result: Status,
+    /// Converted objects in the same order as in the request. Should be empty
+    /// if conversion failed.
+    #[serde(rename = "convertedObjects")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub converted_objects: Option<Vec<serde_json::Value>>,
+}
+
+impl ConversionResponse {
+    /// Creates a new response, matching provided request.
+    /// Returned response is empty: you should call `success` or `error`
+    /// to get a complete `ConversionResponse`.
+    pub fn for_request(request: ConversionRequest) -> Self {
+        ConversionResponse {
+            types: request.types,
+            uid: request.uid,
+            result: Status {
+                status: None,
+                code: 0,
+                message: String::new(),
+                reason: String::new(),
+                details: None,
+            },
+            converted_objects: None,
+        }
+    }
+
+    /// Creates successful conversion response.
+    /// `converted_objects` must specify objects in the exact same order as on input.
+    pub fn success(mut self, converted_objects: Vec<serde_json::Value>) -> Self {
+        self.result = Status::success();
+        self.converted_objects = Some(converted_objects);
+        self
+    }
+
+    /// Creates failed conversion response (discouraged).
+    /// `request_uid` must be equal to the `.uid` field in the request.
+    /// `message` and `reason` will be returned to the apiserver.
+    pub fn error(mut self, message: &str, reason: &str) -> Self {
+        self.result = Status::failure(message, reason);
+        self
+    }
+
+    /// Creates failed conversion response, not matched with any request.
+    /// You should only call this function when request couldn't be parsed into [`ConversionRequest`].
+    /// Otherwise use `error`.
+    pub fn unmatched_error(message: &str, reason: &str) -> Self {
```

Comment:
this can be called `invalid` in line with admission

Reply:
Done